dynamo/plot/scVectorField.py

```diff
@@ -40,8 +42,200 @@
+            raise ValueError(
+                "If you want to plot the vector flow of two genes, please make sure those two genes "
+                "belongs to dynamics genes or .var.use_for_dynamics is True."
+            )
+        X = adata[:, projection_dim_indexer].layers[ekey].A
+        V = adata[:, projection_dim_indexer].layers[vkey].A
+        layer = ekey
+    else:
+        if ("X_" + basis in adata.obsm.keys()) and (vector + "_" + basis in adata.obsm.keys()):
+            X = adata.obsm["X_" + basis][:, projection_dim_indexer]
+            V = adata.obsm[vector + "_" + basis][:, projection_dim_indexer]
+        else:
+            if "X_" + basis not in adata.obsm.keys():
+                layer, basis = basis.split("_")
+                reduceDimension(adata, layer=layer, reduction_method=basis)
+            if "kmc" not in adata.uns_keys():
+                cell_velocities(adata, vkey="velocity_S", basis=basis)
+                X = adata.obsm["X_" + basis][:, projection_dim_indexer]
+                V = adata.obsm[vector + "_" + basis][:, projection_dim_indexer]
+            else:
+                kmc = adata.uns["kmc"]
+                X = adata.obsm["X_" + basis][:, projection_dim_indexer]
+                V = kmc.compute_density_corrected_drift(X, kmc.Idx, normalize_vector=True)
+                adata.obsm[vector + "_" + basis] = V
+
+    X, V = X.copy(), V.copy()
+
+    V /= 3 * quiver_autoscaler(X, V)
+    if inverse:
+        V = -V
+    df = None
+    main_info("X shape: " + str(X.shape) + " V shape: " + str(V.shape))
+    df = pd.DataFrame({"x": X[:, 0], "y": X[:, 1], "z": X[:, 2], "u": V[:, 0], "v": V[:, 1], "w": V[:, 2]})
+
+    if cell_inds == "all":
+        ix_choice = np.arange(adata.shape[0])
+    elif cell_inds == "random":
+        ix_choice = np.random.choice(np.range(adata.shape[0]), size=1000, replace=False)
+    elif type(cell_inds) is int:
+        ix_choice = np.random.choice(np.range(adata.shape[0]), size=cell_inds, replace=False)
+    elif type(cell_inds) is list:
+        if type(cell_inds[0]) is str:
+            cell_inds = [adata.obs_names.to_list().index(i) for i in cell_inds]
+        ix_choice = cell_inds
+
+    df = df.iloc[ix_choice, :]
+
+    if background is None:
+        _background = rcParams.get("figure.facecolor")
+        background = to_hex(_background) if type(_background) is tuple else _background
+
+    head_w, head_l, ax_l, scale = default_quiver_args(quiver_size, quiver_length)
+
+    # single axis output
+    x0, x1, x2 = df.iloc[:, 0], df.iloc[:, 1], df.iloc[:, 2]
+    v0, v1, v2 = df.iloc[:, 3], df.iloc[:, 4], df.iloc[:, 5]
+    nrows = len(color) // ncols
+    if nrows * ncols < len(color):
+        nrows += 1
+    ncols = min(ncols, len(color))
+
+    figure, axes = plt.subplots(nrows, ncols, figsize=figsize, subplot_kw=dict(projection="3d"))
+    axes = np.array(axes)
+    axes_flatten = axes.flatten()
+    for i in range(len(color)):
```

Comment:
add the color argument to this function. 

Also this function is not complete at all. You need to make sure all the arguments of this function are used. Apparently you didn't also plot the scatters in addition to the quiver